pkg/connection/manager.go

```diff
@@ -156,5 +136,81 @@ func (m *DetailsManager) unpublishConnection(ctx context.Context, so SecretOwner
 		Name:     p.Name,
 		Scope:    so.GetNamespace(),
 		Metadata: p.Metadata,
-	}, kv), errDeleteFromStore)
+	}, conn), errDeleteFromStore)
+}
+
+// FetchConnection fetches connection details of a given ConnectionSecretOwner.
+func (m *DetailsManager) FetchConnection(ctx context.Context, so resource.ConnectionSecretOwner) (managed.ConnectionDetails, error) {
+	// This resource does not want to expose a connection secret.
+	p := so.GetPublishConnectionDetailsTo()
+	if p == nil {
+		return nil, nil
+	}
+
+	ss, err := m.connectStore(ctx, p)
+	if err != nil {
+		return nil, errors.Wrap(err, errConnectStore)
+	}
+
+	kv, err := ss.ReadKeyValues(ctx, store.Secret{
+		Name:     p.Name,
+		Scope:    so.GetNamespace(),
+		Metadata: p.Metadata,
+	})
+	return kv, errors.Wrap(err, errReadStore)
+}
+
+// PropagateConnection propagate connection details from one resource to the other.
+func (m *DetailsManager) PropagateConnection(ctx context.Context, to resource.LocalConnectionSecretOwner, from resource.ConnectionSecretOwner) (propagated bool, err error) {
+	// Either from does not expose a connection secret, or to does not want one.
+	if from.GetPublishConnectionDetailsTo() == nil || to.GetPublishConnectionDetailsTo() == nil {
+		return false, nil
+	}
+
+	ssFrom, err := m.connectStore(ctx, from.GetPublishConnectionDetailsTo())
+	if err != nil {
+		return false, errors.Wrap(err, errConnectStore)
+	}
+
+	// TODO(turkenh): Figure out how to check the following case with
+	//  SecretStore: errSecretConflict
+
+	kv, err := ssFrom.ReadKeyValues(ctx, store.Secret{
+		Name:     from.GetPublishConnectionDetailsTo().Name,
+		Scope:    from.GetNamespace(),
+		Metadata: from.GetPublishConnectionDetailsTo().Metadata,
+	})
+	if err != nil {
+		return false, errors.Wrap(err, "errGetSecret")
+	}
+
+	// TODO(turkenh): Implement an equivalent functionality to
+	//  "resource.ConnectionSecretMustBeControllableBy"
```

Comment:
I'm hazy on how this used to work, but I want to make sure that we won't face a security regression if we don't implement this. I think we might have been partly using `ConnectionSecretMustBeControllableBy` to try to avoid malicious folks abusing Crossplane by asking it to overwrite a secret that it shouldn't - for example to propagate `EvilXR`'s connection details to `VictimXRC`'s connection secret.